op-service/eth/types.go

```diff
@@ -228,6 +233,8 @@ type PayloadAttributes struct {
 	PrevRandao Bytes32 `json:"prevRandao"`
 	// suggested value for the coinbase field of the new payload
 	SuggestedFeeRecipient common.Address `json:"suggestedFeeRecipient"`
+	// Withdrawals to include into the block -- should be nil or empty depending on Shanghai enablement
+	Withdrawals *[]Withdrawal `json:"withdrawals,omitempty"`
```

Comment:
Should this be `[]*Withdrawal`?

Reply:
have we tested this pre & post shanghai? geth uses `[]*Withdrawal`, but does require it to be `nil` pre-shanghai and not `nil` post-shanghai

Reply:
> have we tested this pre & post shanghai?

Yeah, I've tested this pre/post. Have some tests similar to the regolith ones that test withdrawals etc are returned post Shanghai. My intention is to submit those as part of the p2p changes PR I'm currently working on.

> geth uses []*Withdrawal, but does require it to be nil pre-shanghai and not nil post-shanghai

That's spot on. This is `*[]Withdrawal` so the omitempty tag works by default pre/post Shanghai. The intention is:

**Pre-Shanghai:**
Withdrawals = nil
`{}`

**Post-Shanghai:**
Withdrawals = []
`{"withdrawals": []}`

I believe doing this without the pointer would require custom serialization based on the canyon timestamp. Happy to do that in EngineClient if ya'll prefer.

Reply:
Sweet. Only question is if we want to do `*[]*Withdrawal`? I feel like `*[]Withdrawal` does work

Reply:
@danyalprout What do you think of the above comment? If you don't want to change it, then you can resolve this comment and this PR will auto merge

Reply:
I don't have a strong opinion either way. My thinking is to leave it as `*[]Withdrawal` for now, as that's what I've tested. If I find any issues when doing the P2P layer, I'll change it as part of that work.